test/remote/gateways/remote_omise_test.rb

```diff
@@ -0,0 +1,140 @@
+require 'test_helper'
+
+class RemoteOmiseTest < Test::Unit::TestCase
+  def setup
+    @gateway = OmiseGateway.new(fixtures(:omise))
+    @amount  = 8888
+    @credit_card   = credit_card('4242424242424242')
+    @declined_card = credit_card('4255555555555555')
+    @invalid_cvc   = credit_card('4024007148673576', {verification_value: ''})
+    @options = {
+      description: 'Active Merchant',
+      email: 'active.merchant@testing.test'
+    }
+  end
+
+  def test_transcript_scrubbing
+    transcript = capture_transcript(@gateway) do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end
+    transcript = @gateway.scrub(transcript)
+
+    assert_scrubbed(@credit_card.number, transcript)
+    assert_scrubbed(@credit_card.verification_value, transcript)
+    assert_scrubbed(@gateway.options[:public_key], transcript)
+  end
+
+  def test_missing_secret_key
+    assert_raise ArgumentError do
+      OmiseGateway.new()
+    end
+  end
+
+  def test_successful_purchase
+    response = @gateway.purchase(@amount, @credit_card)
+    assert_success response
+    assert_equal 'Success', response.message
+    assert_equal response.params['amount'], @amount
+    assert response.params['captured'], 'captured should be true'
+    assert response.params['authorized'], 'authorized should be true'
+  end
+
+  def test_failed_purchase
+    response = @gateway.purchase(@amount, @invalid_cvc)
+    assert_failure response
+    assert_equal Gateway::STANDARD_ERROR_CODE[:invalid_cvc], response.error_code
+  end
+
+  def test_successful_purchase_with_token
+    token = @gateway.send(:omise_tokenize, @credit_card)
```

Comment:
Don't test via hitting `private` methods. Instead, hit the public `store` method, and interact with it.